core/src/test/java/hudson/UtilTest.java

```diff
@@ -259,21 +284,126 @@ public void testIsSymlink() throws IOException, InterruptedException {
         File d = tmp.getRoot();
         try {
             File f = new File(d, "f");
-            OutputStream os = new FileOutputStream(f);
+            // Test: If we cannot delete a file, we throw explaining why
+            mkfiles(f);
+            lockFileForDeletion(f);
             try {
                 Util.deleteFile(f);
                 fail("should not have been deletable");
             } catch (IOException x) {
-                assertEquals(c, x.getClass());
-            } finally {
-                os.close();
+                assertThat(calcExceptionHierarchy(x), hasItem(c));
+                assertThat(x.getMessage(), containsString(f.getPath()));
             }
         } finally {
+            unlockFilesForDeletion();
             Util.deleteRecursive(d);
```

Comment:
I was following the established pattern used in the other unit tests within this class, e.g. line 227.
If I remove my call to deleteRecursive, we have two choices:
(a) I remove it from here but leave it in the other tests, being inconsistent, or
(b) I make further changes to remove this kind of call from the other tests as well (which will make this change larger).

I'm willing to do that extra work if you want, but unless you ask for it, I'll assume that it's better to keep this change as small as possible...